### ISSUE TITLE
Error "Unable to set monitor Port field" when using receiver_creator with smartagent/elasticsearch

### ISSUE THREAD

**Issue by cvauvarin** [signalfx/splunk-otel-collector#2264]
Hi,

I am trying to use the `receiver_creator` feature to create a receiver of type `smartagent/elasticsearch` but the agent is failing with the following error :

```
otelcol[32207]: 2022-11-14T16:53:43.174+0100        error        receivercreator@v0.63.0/observerhandler.go:154        failed to start receiver        {"kind": "receiver", "name": "receiver_creator/1", "pipeline": "metrics", "receiver": "smartagent/elasticsearch", "error": "failed to load template config: unable to set monitor Port field using Endpoint-derived value of 9200: no field Port of type uint16 detected"}
```

My configuration below :

```
    receiver_creator/1:
      watch_observers:
        - host_observer
      receivers:
        smartagent/elasticsearch:
          rule: 'type == "hostport" && port == 9200 && process_name matches "java"'
          config:
            type: elasticsearch
            port: "9200"
            host: 127.0.0.1
            clusterHealthStatsMasterOnly: false
            disableEndpointDimensions: true
            enableEnhancedClusterHealthStats: true
            enableEnhancedNodeIndicesStats:
              - search
              - indexing
              - flush
              - fielddata
              - query_cache
              - request_cache
            extraMetrics:
              - elasticsearch.process.max_file_descriptors
              - elasticsearch.process.cpu.percent
              - elasticsearch.cluster.pending-tasks
              - elasticsearch.cluster.initializing-shards
              - elasticsearch.cluster.status
              - elasticsearch.cluster.task-max-wait-time
              - elasticsearch.jvm.gc.old-time
              - elasticsearch.jvm.gc.count
              - elasticsearch.jvm.mem.heap-used-percent
              - elasticsearch.jvm.mem.pools.old.used_in_bytes
              - elasticsearch.jvm.mem.pools.old.max_in_bytes
              - elasticsearch.jvm.mem.pools.young.used_in_bytes
              - elasticsearch.jvm.mem.pools.young.max_in_bytes
              - elasticsearch.jvm.gc.time
              - elasticsearch.jvm.gc.count
              - elasticsearch.indices.fielddata.evictions
              - elasticsearch.indices.query-cache.evictions
              - elasticsearch.indices.request-cache.evictions
              - elasticsearch.indices.flush.total-time
              - elasticsearch.indices.flush.total
              - elasticsearch.indices.indexing.index-time
              - elasticsearch.indices.indexing.index-total
              - elasticsearch.indices.search.fetch-time
              - elasticsearch.indices.search.fetch-total
```

If we look at the monitor definition in the smartagent (https://github.com/signalfx/signalfx-agent/blob/main/pkg/monitors/elasticsearch/stats/elasticsearch.go#L26), we can see that the `Port` is defined as a string as some others are defined as `uint16` (like mongodb https://github.com/signalfx/signalfx-agent/blob/main/pkg/monitors/collectd/mongodb/mongodb.go#L35). Also if we look as the receiver code the `Port` is always converted as a `uint16` (https://github.com/signalfx/splunk-otel-collector/blob/2fcbfb1f379d3a35f25cf9210f73e8afd135aac3/pkg/receiver/smartagent/config.go#L182).

I dont know if this is an otel collector bug or if the issue should be adressed on the smart agent side if we consider that every monitor Port value must be a uint16.

Can you have a look into this ?

Thanks.

Charles

**Comment by rsicart** [comment]
I have exactly the same problem. Do you have some news about that ?

Thanks in advance.

**Comment by rmfitzpatrick** [comment]
Apologies for the delay and thank you for the report. This will be included in the 0.78.0 release.
